### ISSUE TITLE
eksctl create nodegroup crashes when security groups are set

### ISSUE THREAD

**Issue by drboyer** [weaveworks/eksctl#2006]
**What happened?**
When I ran `eksctl create nodegroup -f <configfile>.yml`, to add a nodegroup to an existing cluster, eksctl crashed and could not add the node group.

This may have something to do with the fact that my cluster and node groups are defined with specific security group settings set.

**What you expected to happen?**
I expected eksctl to create my new nodegroup stack successfully, including only attaching the `sharedNodeSecurityGroup` security group to the launched EC2s.

**How to reproduce it?**
1. Create a cluster with the `vpc.securityGroup` and `vpc.sharedNodeSecurityGroup` settings set (to existing security groups in the configured VPC). The config file can contain a single nodegroup with the following settings set:

```yml
privateNetworking: true
securityGroups:
  withLocal: False
  withShared: True
```

The cluster should come up correctly.

2. Add a new nodegroup to the same config file with the same networking/security group settings as defined above. Try to create it using `eksctl create nodegroup -f <configfile>.yml`.
3. Creation will fail with an "invalid memory address or nil pointer dereference" error (see logs)

Optionally, you can try running `eksctl create nodegroup -f <configfile>.yml` without adding a second nodegroup in step 2. you'd expect eksctl would just say there's nothing to update and exit. Unfortunately, even in this case it attempts to update the cluster and crashes. 

**Anything else we need to know?**
What OS are you using, are you using a downloaded binary or did you compile eksctl, what type of AWS credentials are you using (i.e. default/named profile, MFA) - please don't include actual credentials though!

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.13.0"}
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:20:10Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.9-eks-502bfb", GitCommit:"502bfb383169b124d87848f89e17a04b9fc1f6f0", GitTreeState:"clean", BuildDate:"2020-02-07T01:31:02Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

**Logs**
```
$ eksctl create nodegroup -f ./k8s-configs/cluster-config2.yml -v 4
2020-04-03T18:19:06-04:00 [ℹ]  eksctl version 0.13.0
2020-04-03T18:19:06-04:00 [ℹ]  using region us-east-1
2020-04-03T18:19:06-04:00 [▶]  role ARN for the current session is "arn:aws:sts::234858372212:assumed-role/OperatorRole/devinboyer"
2020-04-03T18:19:07-04:00 [▶]  cluster = {
  Arn: "arn:aws:eks:us-east-1:234858372212:cluster/test-vpc-cluster",
  CertificateAuthority: {
    Data: "REDACTED"
  },
  CreatedAt: 2020-04-03 19:39:44 +0000 UTC,
  Endpoint: "https://<EP>.gr7.us-east-1.eks.amazonaws.com",
  Identity: {
    Oidc: {
      Issuer: "https://oidc.eks.us-east-1.amazonaws.com/id/<EP>"
    }
  },
  Logging: {
    ClusterLogging: [{
        Enabled: false,
        Types: [
          "api",
          "audit",
          "authenticator",
          "controllerManager",
          "scheduler"
        ]
      }]
  },
  Name: "test-vpc-cluster",
  PlatformVersion: "eks.9",
  ResourcesVpcConfig: {
    ClusterSecurityGroupId: "sg-07d2d5556f73a2a1a",
    EndpointPrivateAccess: false,
    EndpointPublicAccess: true,
    PublicAccessCidrs: ["0.0.0.0/0"],
    SecurityGroupIds: ["sg-048fd5d860bf125eb"],
    SubnetIds: ["subnet-0554712596c4079e2","subnet-04e8129d45771e2e2"],
    VpcId: "vpc-0a2694b53cec614de"
  },
  RoleArn: "arn:aws:iam::234858372212:role/eksctl-test-vpc-cluster-cluster-ServiceRole-R6L07OQYPJVY",
  Status: "ACTIVE",
  Tags: {

  },
  Version: "1.14"
}
2020-04-03T18:19:07-04:00 [ℹ]  will use version 1.14 for new nodegroup(s) based on control plane version
2020-04-03T18:19:13-04:00 [▶]  nodegroups = [{
  CreationTime: 2020-04-03 19:50:43.207 +0000 UTC,
  Description: "EKS nodes (AMI family: AmazonLinux2, SSH access: false, private networking: true) [created and managed by eksctl]",
  DisableRollback: false,
  DriftInformation: {
    StackDriftStatus: "NOT_CHECKED"
  },
  EnableTerminationProtection: false,
  Outputs: [
    {
      OutputKey: "FeaturePrivateNetworking",
      OutputValue: "true"
    },
    {
      ExportName: "eksctl-test-vpc-cluster-nodegroup-ng-1-workers::InstanceRoleARN",
      OutputKey: "InstanceRoleARN",
      OutputValue: "arn:aws:iam::234858372212:role/spark-eval-test-roles-NodegroupInstanceRole"
    },
    {
      OutputKey: "FeatureLocalSecurityGroup",
      OutputValue: "false"
    },
    {
      ExportName: "eksctl-test-vpc-cluster-nodegroup-ng-1-workers::InstanceProfileARN",
      OutputKey: "InstanceProfileARN",
      OutputValue: "arn:aws:iam::234858372212:instance-profile/spark-eval-test-roles-NodegroupInstanceProfile"
    },
    {
      OutputKey: "FeatureSharedSecurityGroup",
      OutputValue: "true"
    }
  ],
  RollbackConfiguration: {

  },
  StackId: "arn:aws:cloudformation:us-east-1:234858372212:stack/eksctl-test-vpc-cluster-nodegroup-ng-1-workers/67234760-75e4-11ea-9e6a-0a392ab5a299",
  StackName: "eksctl-test-vpc-cluster-nodegroup-ng-1-workers",
  StackStatus: "CREATE_COMPLETE",
  Tags: [
    {
      Key: "alpha.eksctl.io/cluster-name",
      Value: "test-vpc-cluster"
    },
    {
      Key: "alpha.eksctl.io/nodegroup-name",
      Value: "ng-1-workers"
    },
    {
      Key: "eksctl.cluster.k8s.io/v1alpha1/cluster-name",
      Value: "test-vpc-cluster"
    },
    {
      Key: "Team",
      Value: "DataTools"
    },
    {
      Key: "alpha.eksctl.io/nodegroup-type",
      Value: "unmanaged"
    },
    {
      Key: "CostCategory",
      Value: "rd"
    },
    {
      Key: "eksctl.io/v1alpha2/nodegroup-name",
      Value: "ng-1-workers"
    }
  ]
}]
2020-04-03T18:19:13-04:00 [ℹ]  1 nodegroup(s) that already exist (ng-1-workers) will be excluded
2020-04-03T18:19:13-04:00 [▶]  cfg.json = \
{
    "kind": "ClusterConfig",
    "apiVersion": "eksctl.io/v1alpha5",
    "metadata": {
        "name": "test-vpc-cluster",
        "region": "us-east-1",
        "version": "1.14"
    },
    "iam": {
        "withOIDC": false
    },
    "vpc": {
        "id": "vpc-0a2694b53cec614de",
        "cidr": "192.168.0.0/16",
        "securityGroup": "sg-048fd5d860bf125eb",
        "subnets": {
            "private": {
                "us-east-1b": {
                    "id": "subnet-0554712596c4079e2",
                    "cidr": "192.168.96.0/19"
                },
                "us-east-1f": {
                    "id": "subnet-04e8129d45771e2e2",
                    "cidr": "192.168.64.0/19"
                }
            }
        },
        "sharedNodeSecurityGroup": "sg-071867f9dc75e2c95",
        "clusterEndpoints": {
            "privateAccess": false,
            "publicAccess": true
        }
    },
    "availabilityZones": [
        "us-east-1f",
        "us-east-1b"
    ],
    "status": {
        "endpoint": "https://<EP>.gr7.us-east-1.eks.amazonaws.com",
        "certificateAuthorityData": "REDACTED",
        "arn": "arn:aws:eks:us-east-1:234858372212:cluster/test-vpc-cluster"
    }
}

2020-04-03T18:19:16-04:00 [▶]  ClusterSharedNodeSecurityGroup = sg-071867f9dc75e2c95
2020-04-03T18:19:16-04:00 [ℹ]  combined exclude rules: ng-1-workers
2020-04-03T18:19:16-04:00 [ℹ]  1 nodegroup (ng-1-workers) was excluded (based on the include/exclude rules)
2020-04-03T18:19:16-04:00 [ℹ]  2 sequential tasks: { fix cluster compatibility, no tasks }
2020-04-03T18:19:16-04:00 [▶]  started task: fix cluster compatibility
2020-04-03T18:19:16-04:00 [ℹ]  checking cluster stack for missing resources
2020-04-03T18:19:20-04:00 [ℹ]  cluster stack is missing resources for Managed Nodegroups
2020-04-03T18:19:20-04:00 [ℹ]  adding missing resources to cluster stack
2020-04-03T18:19:20-04:00 [ℹ]  re-building cluster stack "eksctl-test-vpc-cluster-cluster"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x22934d1]

goroutine 114 [running]:
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).addOutputsForVPC(0xc0000e00f0)
	/src/pkg/cfn/builder/vpc.go:166 +0x101
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).AddAllResources(0xc0000e00f0, 0xc0002825a0, 0xc0006d3130)
	/src/pkg/cfn/builder/cluster.go:68 +0xac
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).AppendNewClusterStackResource(0xc000aceb70, 0x100, 0x0, 0x0, 0x0)
	/src/pkg/cfn/manager/cluster.go:116 +0x362
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).FixClusterCompatibility(0xc000aceb70, 0xc000677ee8, 0x100493d)
	/src/pkg/cfn/manager/compat.go:64 +0x342
github.com/weaveworks/eksctl/pkg/cfn/manager.(*clusterCompatTask).Do(0xc000bb6500, 0xc000644c00, 0x0, 0x0)
	/src/pkg/cfn/manager/tasks.go:190 +0x6b
github.com/weaveworks/eksctl/pkg/cfn/manager.doSingleTask(0xc000644b40, 0x414f7c0, 0xc000bb6500, 0x0)
	/src/pkg/cfn/manager/tasks.go:265 +0xec
github.com/weaveworks/eksctl/pkg/cfn/manager.doSequentialTasks(0xc000644b40, 0xc000bb6580, 0x2, 0x2)
	/src/pkg/cfn/manager/tasks.go:295 +0x70
created by github.com/weaveworks/eksctl/pkg/cfn/manager.(*TaskTree).DoAllSync
	/src/pkg/cfn/manager/tasks.go:118 +0x237
```

**Comment by drboyer** [comment]
FWIW this seems similar to #1160, but with a different message immediately before the crash (here it's talking about "missing resources for Managed Nodegroups").

**Comment by cPu1** [comment]
This has been fixed in 0.16 by https://github.com/weaveworks/eksctl/pull/1908. Please try it out and let us know if you're still getting the error.

**Comment by martina-if** [comment]
I am closing this since it seems it has been resolved already. Please feel free to re-open it if that's not the case.
